gradio/route_utils.py

```diff
@@ -937,3 +937,13 @@ async def add_segment(self, data: MediaStreamChunk | None):
 
     def end_stream(self):
         self.ended = True
+
+
+def create_url_safe_hash(data: bytes, digest_size=8):
+    """Create a URL-safe short hash of the data. Used to generate unique short deep links."""
+    import base64
+
+    hash_obj = hashlib.blake2b(data, digest_size=digest_size)
```

Comment:
for https://github.com/gradio-app/gradio/issues/10400
```suggestion
    hash_obj = hashlib.blake2b(data, digest_size=digest_size, usedforsecurity=False)
```

Reply:
Wow good call - was not on my radar at all.

Reply:
We should add a semgrep test for this